src/Avalonia.Controls.DataGrid/DataGrid.cs

```diff
@@ -4161,7 +4161,7 @@ void SetValidationStatus(ICellEditBinding binding)
 
             ResetValidationStatus();
 
-            if (exitEditingMode)
+            if (exitEditingMode && CurrentColumn != null)
```

Comment:
I think this change should be included in a separate PR where the reason of change is explained.